File: src/main/java/com/mrpowergamerbr/loritta/commands/vanilla/magic/NashornTestCommand.java
package com.mrpowergamerbr.loritta.commands.vanilla.magic;

import com.mrpowergamerbr.loritta.Loritta;
import com.mrpowergamerbr.loritta.commands.CommandBase;
import com.mrpowergamerbr.loritta.commands.CommandCategory;
import com.mrpowergamerbr.loritta.commands.CommandContext;
import com.mrpowergamerbr.loritta.commands.nashorn.NashornCommand;

public class NashornTestCommand extends CommandBase {
	@Override public String getLabel() {
		return "nashorn";
	}

	@Override public CommandCategory getCategory() {
		return CommandCategory.MAGIC;
	}

	@Override public void run(CommandContext context) {
		if (context.getUserHandle().getId().equals(Loritta.getConfig()
				.getOwnerId())) { // Somente o dono do bot pode usar eval! Eval é uma ferramenta muito poderosa que não deve ser usada por qualquer um!
			String javaScript = String.join(" ", context.getArgs());

			NashornCommand nashornCmd = new NashornCommand("teste", javaScript);

			System.out.println("Executando NashornCommand...");

			nashornCmd.handle(context.getEvent(), context.getConfig().commandPrefix + "teste", context.getConfig());
		} else {
			// Sem permissão
		}
	}
}

File: src/main/java/com/mrpowergamerbr/loritta/commands/vanilla/magic/NashornTestCommand.kt
package com.mrpowergamerbr.loritta.commands.vanilla.magic

import com.mrpowergamerbr.loritta.Loritta
import com.mrpowergamerbr.loritta.commands.AbstractCommand
import net.perfectdreams.loritta.api.commands.CommandCategory
import com.mrpowergamerbr.loritta.commands.CommandContext
import com.mrpowergamerbr.loritta.commands.nashorn.NashornCommand
import com.mrpowergamerbr.loritta.utils.locale.LegacyBaseLocale

class NashornTestCommand : AbstractCommand("nashorn", category = CommandCategory.MAGIC) {
	override fun getDescription(locale: LegacyBaseLocale): String {
		return "Executa códigos em JavaScript usando a sandbox de comandos da Loritta"
	}

	override fun canHandle(context: CommandContext): Boolean {
		return context.userHandle.id in Loritta.config.subOwnerIds || context.userHandle.id == Loritta.config.ownerId
	}

	override suspend fun run(context: CommandContext,locale: LegacyBaseLocale) {
		val javaScript = context.args.joinToString(" ")

		val nashornCmd = NashornCommand("teste", javaScript)

		nashornCmd.run(context, locale)
	}
}